testutil/ciao-down/ciao_down.go

```diff
@@ -0,0 +1,265 @@
+		return
+	}
+
+	fmt.Println("Installing host dependencies")
+	installDeps(ctx)
+
+	err = os.MkdirAll(ws.instanceDir, 0755)
+	if err != nil {
+		errCh <- fmt.Errorf("unable to create cache dir: %v", err)
+		return
+	}
+
+	failed := true
+	defer func() {
+		if failed {
+			_ = os.RemoveAll(ws.instanceDir)
+		}
+	}()
+
+	qcowPath, err := downloadUbuntu(ctx, ws.ciaoDir, downloadProgress)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = buildISOImage(ctx, ws.instanceDir, ws)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = createRootfs(ctx, qcowPath, ws.instanceDir)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = bootVM(ctx, ws)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = manageInstallation(ctx, ws.instanceDir, ws)
+	if err != nil {
+		errCh <- err
+		return
+	}
+	errCh <- nil
+	failed = false
+	fmt.Println("VM successfully created!")
+	fmt.Println("Type ciao-down connect to start using it.")
+}
+
+func start(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = bootVM(ctx, ws)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	if err == nil {
+		fmt.Println("VM Started")
+	}
+
+	errCh <- err
+}
+
+func stop(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = stopVM(ctx, ws.instanceDir)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	if err == nil {
+		fmt.Println("VM Stopped")
+	}
+
+	errCh <- err
+}
+
+func quit(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	err = quitVM(ctx, ws.instanceDir)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	if err == nil {
+		fmt.Println("VM Quit")
+	}
+
+	errCh <- err
+}
+
+func status(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	statusVM(ctx, ws.instanceDir)
+	errCh <- err
+}
+
+func connect(errCh chan error) {
+	path, err := exec.LookPath("ssh")
+	if err != nil {
+		errCh <- fmt.Errorf("Unable to locate ssh binary")
+	}
+
+	err = syscall.Exec(path, []string{path, "127.0.0.1", "-p", "10022"},
+		os.Environ())
+	errCh <- err
+}
+
+func delete(ctx context.Context, errCh chan error) {
+	ws, err := prepareEnv(ctx)
+	if err != nil {
+		errCh <- err
+		return
+	}
+
+	_ = quitVM(ctx, ws.instanceDir)
+	err = os.RemoveAll(ws.instanceDir)
+	if err != nil {
+		errCh <- fmt.Errorf("unable to delete instance: %v", err)
+		return
+	}
+
+	knownHosts := path.Join(ws.Home, ".ssh", "known_hosts")
+	err = exec.Command("ssh-keygen", "-f", knownHosts, "-R", "[127.0.0.1]:10022").Run()
+	if err != nil {
+		fmt.Println("Failed to remove VM entry from known_hosts")
+	}
+
+	errCh <- nil
+}
+
+func runCommand(signalCh <-chan os.Signal) error {
+	var err error
+
+	errCh := make(chan error)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	switch os.Args[1] {
```

Comment:
I meant to say main package's os.Arg's, but that then answers the question right away itself.  Arg's are in the os package and not the main package or tied to its main() function.

It sure feels weird if any piece of code anywhere could check if os.Args[42] is set and use it if so.